src/senaite/sync/importstep.py

```diff
@@ -185,14 +186,22 @@ def _import_data(self):
 
         for item_count, r_uid in enumerate(ordered_uids):
             row = self.sh.find_unique("remote_uid", r_uid)
-            logger.info("Handling: {} ".format(row["path"]))
+            logger.debug("Handling: {} ".format(row["path"]))
             self._handle_obj(row)
 
             # Handling object means there is a chunk containing several objects
             # which have been created and updated. Reindex them now.
             self.uids_to_reindex = list(set(self.uids_to_reindex))
             for uid in self.uids_to_reindex:
-                api.get_object_by_uid(uid).reindexObject()
+                # It is possible that the object has a method (not a Field
+                # in its Schema) which is used as an index and it fails.
+                # TODO: Make sure reindexing won't fail!
+                try:
+                    api.get_object_by_uid(uid).reindexObject()
+                except Exception, e:
```

Comment:
Your exception catch is too generic in here.
E.g. it swallows the case when `api.get_object_by_uid` fails with or your `reindexObject` fails.
Try to divide this into two steps and try to find out which errors occur. The more specific we are, the better we can debug it later

Reply:
you are right that just `Exception` is not good at all, but 'reindexing' can fail with too many error types. I added that 'TODO' there to find out which method (that is not a field in the Scehma) of the object can fail so we actually do not need this error handling here... For now, to be safe I think it is better to have generic Exception case, if you don't mind.